utils.py

```diff
@@ -5,33 +5,142 @@
-#     sfreq = sample rate
-#     ch_types = channel type, doesn't seem to support raw optical, leave empty
-#     montage = can be a string or list containing channel positions
-#     verbose = I assume for logging stuff
-# =============================================================================
+import mne
+import numpy as np
+import pandas as pd
+
+def boxy2mne(boxy_file,mtg_file,tol_file):
     
 # =============================================================================
-# so we currently load the participant montage and their tol info
-# montage file contains (likely) a subset of the channels in the tol file
-# so what we need to do if compare our montage channels and find the 
-# coordinates from the tol
+#     ready raw data from boxy file
 # =============================================================================
+    ###this keeps track of the line we're on###
+    ###mostly to know the start and stop of data (probably an easier way)###
+    line_num = 0
+    
+    ###load and read data to get some meta information###
+    ###there is alot of information at the beginning of a file###
+    ###but this only grabs some of it###
+    with open(boxy_file,'r') as data:
+        for i_line in data:
+            line_num += 1
+            if '#DATA ENDS' in i_line:
+                end_line = line_num - 1
+                break
+            if 'Detector Channels' in i_line:
+                detect_num = int(i_line.rsplit(' ')[0])
+            elif 'External MUX Channels' in i_line:
+                mux_num = int(i_line.rsplit(' ')[0])
+            elif 'Auxiliary Channels' in i_line:
+                aux_num = int(i_line.rsplit(' ')[0])
+            elif 'Waveform (CCF) Frequency (Hz)' in i_line:
+                ccf_ha = float(i_line.rsplit(' ')[0])
+            elif 'Update Rate (Hz)' in i_line:
+                srate = float(i_line.rsplit(' ')[0])
+            elif 'Updata Rate (Hz)' in i_line:
+                srate = float(i_line.rsplit(' ')[0])
+            elif '#DATA BEGINS' in i_line:
+                start_line = line_num
 
-import mne
-
-def boxy2mne(mtg_file,tol_file):
+    ###now let's go through and parse our raw data###  
+    raw_data = pd.read_csv(boxy_file, skiprows=start_line, sep='\t')
     
-    ###need to have function determine sampling rate automatically, eventually
-    srate = 39.0625
+    ###only go up to 'J' since that the max number of detectors we can have###
+    ###for now anyway###
+    detectors = ['A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L', 
+                 'M', 'N', 'O', 'P', 'Q', 'R', 'S', 'T', 'U', 'V', 'W', 'X',
+                 'Y', 'Z']
+    data_types = ['AC','DC','Ph']
+    sources = np.arange(1,mux_num+1,1)
+    
+    ###since we can save boxy files in two different styles###
+    ###this will check to see which style the data is saved###
+    ###seems to also work with older boxy files###
+    if 'exmux' in raw_data.columns:
+        filetype = 1
```

Comment:
can you make a better name for these two filetypes (e.g. 'imagent2', 'newboxy')

Reply:
okay, changed to be 'parsed' and 'non-parsed' instead of '1' or '2'